services/web/server/tests/unit/with_dbs/01/test_api_keys.py

```diff
@@ -215,3 +219,29 @@ async def test_get_not_existing_api_key(
 
     if not errors:
         assert data is None
+
+
+@pytest.fixture
+async def app_environment(
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+) -> EnvVarsDict:
+    webserver_garbage_collector = '{"GARBAGE_COLLECTOR_INTERVAL_S": 30, "GARBAGE_COLLECTOR_PRUNE_APIKEYS_INTERVAL_S": 1}'
+    monkeypatch.setenv("WEBSERVER_GARBAGE_COLLECTOR", webserver_garbage_collector)
+    return app_environment | {
+        "WEBSERVER_GARBAGE_COLLECTOR": webserver_garbage_collector
+    }
+
+
+async def test_prune_expired_api_keys_task_is_triggered(
+    app_environment: EnvVarsDict, mocker: MockerFixture, client: TestClient
+):
+    mock = mocker.patch(
+        "simcore_service_webserver.api_keys._service._repository.prune_expired"
+    )
+    settings = client.server.app[  # type: ignore
+        APP_SETTINGS_KEY
+    ].WEBSERVER_GARBAGE_COLLECTOR
+    assert isinstance(settings, GarbageCollectorSettings)
+    await asyncio.sleep(2 * settings.GARBAGE_COLLECTOR_PRUNE_APIKEYS_INTERVAL_S)
```

Comment:
here another tecnique is to retry **with timeout** until called.
The also one could called whether the time to call fits accurately the interval

Reply:
yes please use tenacity here.

Reply:
Please do use tenacity here